src/ophyd_async/core/_hdf_dataset.py

```diff
@@ -79,17 +78,19 @@ def __init__(
             for ds in datasets
         ]
 
-    def stream_resources(self) -> Iterator[StreamResource]:
-        for bundle in self._bundles:
-            yield bundle.stream_resource_doc
+    async def collect_stream_docs(
+        self, indices_written: int
+    ) -> AsyncIterator[StreamAsset]:
```

Comment:
Ok, having seen the code it does no I/O, so we can put it back to a plain iterator

```suggestion
    def collect_stream_docs(
        self, indices_written: int
    ) -> Iterator[StreamAsset]:
```